Src/Plugins/Security/SEC101_050.MongoDbConnectionStringValidator.cs

```diff
@@ -0,0 +1,99 @@
+{
+    public class MongoDbConnectionStringValidator : ValidatorBase
+    {
+        internal static MongoDbConnectionStringValidator Instance;
+
+        static MongoDbConnectionStringValidator()
+        {
+            Instance = new MongoDbConnectionStringValidator();
+        }
+
+        public static string IsValidStatic(ref string matchedPattern,
+                                           ref Dictionary<string, string> groups,
+                                           ref string failureLevel,
+                                           ref string fingerprint,
+                                           ref string message)
+        {
+            return IsValidStatic(Instance,
+                                 ref matchedPattern,
+                                 ref groups,
+                                 ref failureLevel,
+                                 ref fingerprint,
+                                 ref message);
+        }
+
+        public static string IsValidDynamic(ref string fingerprint, ref string message, ref Dictionary<string, string> options)
+        {
+            return IsValidDynamic(Instance,
+                                  ref fingerprint,
+                                  ref message,
+                                  ref options);
+        }
+
+        protected override string IsValidStaticHelper(ref string matchedPattern,
+                                                      ref Dictionary<string, string> groups,
+                                                      ref string failureLevel,
+                                                      ref string fingerprintText,
+                                                      ref string message)
+        {
+            if (!groups.TryGetNonEmptyValue("account", out string account) ||
+                !groups.TryGetNonEmptyValue("password", out string password) ||
+                !groups.TryGetNonEmptyValue("host", out string host))
+            {
+                return nameof(ValidationState.NoMatch);
+            }
+
+            fingerprintText = new Fingerprint()
+            {
+                Account = account,
+                Password = password,
+                Host = host,
+            }.ToString();
+
+            return nameof(ValidationState.Unknown);
+        }
+
+        protected override string IsValidDynamicHelper(ref string fingerprintText,
+                                                       ref string message,
+                                                       ref Dictionary<string, string> options)
+        {
+            var fingerprint = new Fingerprint(fingerprintText, false);
+
+            string host = fingerprint.Host;
+            string account = fingerprint.Account;
+            string password = fingerprint.Password;
+
+            try
+            {
+                var dbClient = new MongoClient($"mongodb+srv://{account}:{password}@{host}/?connectTimeoutMS=3000");
```

Comment:
>dbClient [](start = 20, length = 8)

Just curious, are you sure this type isn't disposable? thanks for adding a timeout!